### PR TITLE
Enhancements for Better Accessibility and Code Refactoring ⚙

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
This PR introduces minor improvements, such as:

1. Adding the `type="search"` attribute to the input tag for better accessibility.
2. Adding decoding properties in the img tag.
3. Refactoring the code by adding event modifiers directly using '|'.

Extremely sorry if I made any mistakes :(

### CODE REVIEW

File: pagefind_ui/default/svelte/ui.svelte
Comment (itsmatteomanf) [inline pagefind_ui/default/svelte/ui.svelte:205]: This would be my only issue, I'd keep it as before, as you are now preventing default for *all* `keydown` events, which wasn't the case before.

```suggestion
            on:keydown={(e) => {
```

File: pagefind_ui/default/svelte/ui.svelte
Comment (itsmatteomanf) [inline pagefind_ui/default/svelte/ui.svelte:209]:
```suggestion
                }
                if (e.key === "Enter") {
                     e.preventDefault();
                }
```

**Review by itsmatteomanf — CHANGES_REQUESTED** [review]
I'd just revert the `preventDefault` I commented on.

File: pagefind_ui/default/svelte/ui.svelte
Comment (sanjaiyan-dev) [reply pagefind_ui/default/svelte/ui.svelte]: > This would be my only issue, I'd keep it as before, as you are now preventing default for _all_ `keydown` events, which wasn't the case before.

Done :)

File: pagefind_ui/default/svelte/ui.svelte
Comment (itsmatteomanf) [inline pagefind_ui/default/svelte/ui.svelte:225]: Extra space, but it's a *really* minor thing.

```suggestion
                    e.preventDefault();
```

**Review by itsmatteomanf — COMMENTED** [review]
LGTM, but not sure if all these changes are correct. I'm gonna wait for a second approval.

**Comment by sanjaiyan-dev** [comment]
🧐🫠

**Comment by bglw** [comment]
Sorry to leave you hanging! Had to deal with some tasks that were put aside while working on the 1.0 release. Coming back around to new PRs and issues in this repo soon.

**Comment by bglw** [comment]
Hi @sanjaiyan-dev — on Chrome (and others?) an input with `type="search"` shows a clear button in the input. Since the Pagefind UIs have their own implementations of this, it would be good to hide the browser default.

Let me know if this is a change you're keen to make or if you'd rather I tackle it 🙂

Screenshot 2023-10-13 at 3 14 54 PM

**Comment by sanjaiyan-dev** [comment]
> Hi @sanjaiyan-dev — on Chrome (and others?) an input with `type="search"` shows a clear button in the input. Since the Pagefind UIs have their own implementations of this, it would be good to hide the browser default.
> 
> Let me know if this is a change you're keen to make or if you'd rather I tackle it 🙂 Everything else looks good!
> 
> Screenshot 2023-10-13 at 3 14 54 PM

@bglw 
If you believe that adding the 'type=search' attribute may not be necessary, I would like to suggest considering the use of the 'inputmode' attribute as an alternative. You can find more information about it on the Mozilla Developer Network here: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode

**Comment by bglw** [comment]
I think the type search is good — we just need to normalize the styling. At a glance, https://stackoverflow.com/a/77190241 seems good, but will need to test it across browsers to make sure

**Comment by sanjaiyan-dev** [comment]
> I think the type search is good — we just need to normalize the styling. At a glance, https://stackoverflow.com/a/77190241 seems good, but will need to test it across browsers to make sure

🤝

**Pull request closed without merging**
